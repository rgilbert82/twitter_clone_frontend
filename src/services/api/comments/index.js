export { default as getCommentsAPI } from './getComments.js';
export { default as createCommentAPI } from './createComment.js';
export { default as editCommentAPI } from './editComment.js';
export { default as deleteCommentAPI } from './deleteComment.js';
