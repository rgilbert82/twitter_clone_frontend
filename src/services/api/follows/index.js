export { default as getFollowAPI } from './getFollow.js';
export { default as createFollowAPI } from './createFollow.js';
export { default as deleteFollowAPI } from './deleteFollow.js';
export { default as getUserFollowersAPI } from './getUserFollowers.js';
export { default as getUserFollowingsAPI } from './getUserFollowings.js';
