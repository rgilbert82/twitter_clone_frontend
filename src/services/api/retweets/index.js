export { default as createRetweetAPI } from './createRetweet.js';
export { default as deleteRetweetAPI } from './deleteRetweet.js';
