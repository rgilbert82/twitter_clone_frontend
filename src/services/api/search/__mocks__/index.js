export { default as getSearchResultsAPI } from './getSearchResults.js';
