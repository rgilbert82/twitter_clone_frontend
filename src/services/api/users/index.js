export { default as getUsersAPI } from './getUsers.js';
export { default as getUserAPI } from './getUser.js';
export { default as createUserAPI } from './createUser.js';
export { default as updateUserAPI } from './updateUser.js';
export { default as getUserTweetsAPI } from './getUserTweets.js';
export { default as deleteUserAPI } from './deleteUser.js';
