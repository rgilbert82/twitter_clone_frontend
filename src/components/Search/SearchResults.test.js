jest.mock('../../services/api/search');

import React from 'react';
import { shallow } from 'enzyme';
import { SearchResults } from '.';
import { getSearchResultsAPI } from '../../services/api/search';

describe(SearchResults, () => {
  const currentUser = { id: 1, name: 'Ryan', username: 'RG', token: '12345', slug: 'rg', followers: 1, following: 1, tweet_count: 1, retweet_count: 1, all_tweet_count: 2 };
  let fetchSpy = jest.spyOn(SearchResults.prototype, 'fetchResults');
  let component;

  beforeEach(() => {
    component = shallow(<SearchResults match={{params: { search_field: 'sports'}}} loggedIn={true} currentUser={currentUser} displayMessage={jest.fn()} />);
  });

  afterAll(() => {
    fetchSpy.mockClear();
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("triggers fetchResults on mount", () => {
    expect(fetchSpy).toHaveBeenCalled();
  });

  it("fetches the tweets", () => {
    expect.assertions(2);
    component.instance().fetchResults()
      .then(() => {
        expect(component.state('results')).toEqual([{ id: 1, body: 'something here', user_id: 5, image: null }]);
        expect(component.state('resultsLoaded')).toEqual(true);
      });
  });
});
