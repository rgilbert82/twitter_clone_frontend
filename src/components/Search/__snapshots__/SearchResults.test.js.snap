// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class SearchResults extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      results: [],
      resultsFeed: [],
      resultsLoaded: false };


    this.fetchResults = this.fetchResults.bind(this);
    this.scrollListener = this.scrollListener.bind(this);
  }

  componentDidMount() {
    this._isMounted = true;
    if (this._isMounted) {
      this.fetchResults();
      window.addEventListener('scroll', this.scrollListener);
    }
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.scrollListener);
    this._isMounted = false;
  }

  componentDidUpdate(prevProps, prevState) {
    const elements = document.getElementsByClassName('tweet_body_text');

    if (prevProps.location.pathname !== this.props.location.pathname) {
      this.setState({ results: [], resultsLoaded: false }, () => {
        this.fetchResults();
      });
    }

    for (let i = 0; i < elements.length; i++) {
      elements[i].innerHTML = (0, _misc.hashtagify)(elements[i].innerHTML);
    }
  }

  scrollListener() {
    if (window.innerHeight + window.pageYOffset >= document.body.offsetHeight - 2) {
      if (this.state.resultsFeed.length < this.state.results.length) {
        this.setState({ resultsFeed: this.state.results.slice(0, this.state.resultsFeed.length + 8) });
      }
    }
  }

  fetchResults() {
    const searchField = this.props.match.params.search_field;

    return (0, _search.getSearchResultsAPI)(searchField).
    then(data => {
      this.setState({ results: data, resultsFeed: data.slice(0, 8), resultsLoaded: true });
    }).catch(err => {
      console.log(err);
    });
  }

  render() {
    let pluralize;
    let allTweets;
    let content;

    if (this.state.resultsLoaded && this.state.results.length === 0) {
      content =
      _react2.default.createElement('div', { id: 'section_message', __source: { fileName: _jsxFileName, lineNumber: 73 }, __self: this },
        _react2.default.createElement('h2', { __source: { fileName: _jsxFileName, lineNumber: 74 }, __self: this }, 'Your search returned 0 results'));


    } else if (this.state.resultsLoaded) {
      allTweets = this.state.resultsFeed.map(tweet => {
        return (
          _react2.default.createElement('li', { className: 'list_item', key: tweet.id, __source: { fileName: _jsxFileName, lineNumber: 80 }, __self: this },
            _react2.default.createElement(_Tweets.TweetListItem, { history: this.props.history, navigateToSearch: this.props.navigateToSearch, tweet: tweet, loggedIn: this.props.loggedIn, displayMessage: this.props.displayMessage, __source: { fileName: _jsxFileName, lineNumber: 81 }, __self: this })));


      });

      if (this.state.results.length !== 1) {
        pluralize = "s";
      } else {
        pluralize = "";
      }

      content =
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 93 }, __self: this },
        _react2.default.createElement('h3', { className: 'search_results_header', __source: { fileName: _jsxFileName, lineNumber: 94 }, __self: this }, '"#', this.props.match.params.search_field, '" returned ', this.state.results.length, ' result', pluralize),
        _react2.default.createElement('ul', { __source: { fileName: _jsxFileName, lineNumber: 95 }, __self: this },
          allTweets));



    } else {
      content = _react2.default.createElement(_Misc.PageLoading, { __source: { fileName: _jsxFileName, lineNumber: 101 }, __self: this });
    }

    return _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 104 }, __self: this }, content);
  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    currentUser={
        Object {
            "all_tweet_count": 2,
            "followers": 1,
            "following": 1,
            "id": 1,
            "name": "Ryan",
            "retweet_count": 1,
            "slug": "rg",
            "token": "12345",
            "tweet_count": 1,
            "username": "RG",
          }
    }
    displayMessage={[Function]}
    loggedIn={true}
    match={
        Object {
            "params": Object {
              "search_field": "sports",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <PageLoading />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <PageLoading />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
